src/Symfony/Component/DependencyInjection/Compiler/CheckTypeDeclarationsPass.php

```diff
@@ -174,12 +178,24 @@ private function checkType(Definition $checkedDefinition, $value, \ReflectionPar
             throw new InvalidParameterTypeException($this->currentId, $class, $parameter);
         }
 
-        if ($value instanceof Parameter) {
-            $value = $this->container->getParameter($value);
-        } elseif ($value instanceof Expression) {
+        if ($value instanceof Expression) {
             $value = $this->getExpressionLanguage()->evaluate($value, ['container' => $this->container]);
-        } elseif (\is_string($value) && '%' === ($value[0] ?? '') && preg_match('/^%([^%]+)%$/', $value, $match)) {
-            $value = $this->container->getParameter($match[1]);
+        } elseif (\is_string($value)) {
+            if ('%' === ($value[0] ?? '') && preg_match('/^%([^%]+)%$/', $value, $match)) {
+                // Only array parameters are not inlined when dumped.
```

Comment:
Added comments because those things are kind of hard to follow. We actually don't need to resolve the value here since we only care about the type.